File: 238-product-of-array-except-self/product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    //length := len(nums)
    res:= make([]int, len(nums))
    prefix := 1
    for i := range nums{
        res[i] = prefix
        prefix *= nums[i]
    }
    postfix := 1
    for i := len(nums) - 1; i > -1; i--{
        res[i] *= postfix
        postfix *= nums[i]
    }
    return res
}

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        test = []
        prefix = 1
        for i in range(len(nums)):
            test.append(prefix)
            prefix *= nums[i]
            
        postfix = 1
        for i in range(len(nums)-1, -1, -1):
            test[i] *= postfix
            postfix *= nums[i]
        return(test)